wled00/FX_2Dfcn.cpp

```diff
@@ -273,119 +286,62 @@ uint32_t IRAM_ATTR_YN Segment::getPixelColorXY(int x, int y) const {
-      } else setPixelColorXY(x, row, curnew); // first pixel
-      lastnew = curnew;
-      last = cur; // save original value for comparison on next iteration
-      carryover = part;
+  if(blur_x) {
+    const uint8_t keepx = smear ? 255 : 255 - blur_x;
+    const uint8_t seepx = blur_x >> (1 + smear);
+    for (unsigned row = 0; row < rows; row++) { // blur rows (x direction)
+      uint32_t carryover = BLACK;
+      uint32_t curnew = BLACK;
+      for (unsigned x = 0; x < cols; x++) {
+        uint32_t cur = getPixelColorXY(x, row);
+        uint32_t part = color_fade(cur, seepx);
+        curnew = color_fade(cur, keepx);
+        if (x > 0) {
+          if (carryover) curnew = color_add(curnew, carryover);
+          uint32_t prev = color_add(lastnew, part);
+          // optimization: only set pixel if color has changed
+          if (last != prev) setPixelColorXY(x - 1, row, prev);
+        } else setPixelColorXY(x, row, curnew); // first pixel
+        lastnew = curnew;
+        last = cur; // save original value for comparison on next iteration
+        carryover = part;
+      }
+      setPixelColorXY(cols-1, row, curnew); // set last pixel
     }
-    setPixelColorXY(cols-1, row, curnew); // set last pixel
   }
-  for (unsigned col = 0; col < cols; col++) {
-    uint32_t carryover = BLACK;
-    uint32_t curnew = BLACK;
-    for (unsigned y = 0; y < rows; y++) {
-      uint32_t cur = getPixelColorXY(col, y);
-      uint32_t part = color_fade(cur, seep);
-      curnew = color_fade(cur, keep);
-      if (y > 0) {
-        if (carryover) curnew = color_add(curnew, carryover);
-        uint32_t prev = color_add(lastnew, part);
-        // optimization: only set pixel if color has changed
-        if (last != prev) setPixelColorXY(col, y - 1, prev);
-      } else setPixelColorXY(col, y, curnew); // first pixel
-      lastnew = curnew;
-      last = cur; //save original value for comparison on next iteration
-      carryover = part;
+  if(blur_y) {
+    const uint8_t keepy = smear ? 255 : 255 - blur_y;
+    const uint8_t seepy = blur_y >> (1 + smear);
+    for (unsigned col = 0; col < cols; col++) {
+      uint32_t carryover = BLACK;
+      uint32_t curnew = BLACK;
+      for (unsigned y = 0; y < rows; y++) {
+        uint32_t cur = getPixelColorXY(col, y);
+        uint32_t part = color_fade(cur, seepy);
+        curnew = color_fade(cur, keepy);
+        if (y > 0) {
+          if (carryover) curnew = color_add(curnew, carryover);
+          uint32_t prev = color_add(lastnew, part);
+          // optimization: only set pixel if color has changed
+          if (last != prev) setPixelColorXY(col, y - 1, prev);
+        } else setPixelColorXY(col, y, curnew); // first pixel
+        lastnew = curnew;
+        last = cur; //save original value for comparison on next iteration
+        carryover = part;
+      }
+      setPixelColorXY(col, rows - 1, curnew);
     }
-    setPixelColorXY(col, rows - 1, curnew);
   }
 }
 
+/*
```

Comment:
I know it is unused ATM but it produces quite different blur and I put this function in intentionally.
I would not like it removed.